runtime/testy/runtime.py

```diff
@@ -34,43 +34,64 @@ def runtime(testName=""):
   testMode = testName != ""
   maxIter = 3 if testMode else None
 
+  def nonTestMode_print(*args):
+    """Prints only if we are NOT in testMode"""
+    if not testMode:
+      print(args)
+
   badThingsQueue = multiprocessing.Queue()
   stateQueue = multiprocessing.Queue()
   spawnProcess = processFactory(badThingsQueue, stateQueue)
   restartCount = 0
   emergency_stopped = False
+
   try:
     spawnProcess(PROCESS_NAMES.STATE_MANAGER, startStateManager)
     spawnProcess(PROCESS_NAMES.UDP_SEND_PROCESS, startUDPSender)
     spawnProcess(PROCESS_NAMES.UDP_RECEIVE_PROCESS, startUDPReceiver)
     spawnProcess(PROCESS_NAMES.HIBIKE, startHibike)
+    controlState = "idle"
+
     while True:
-      if not testMode:
-        if restartCount >= 3:
-          print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
-          print("Too many restarts, terminating")
-          break
-        if emergency_stopped:
-          print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
-          print("terminating due to E-Stop")
-          break
-        print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
-        print("Starting studentCode attempt: %s" % (restartCount,))
-      spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode, testName, maxIter)
+      if testMode:
+        # Automatically enter telop mode when running tests
+        badThingsQueue.put(BadThing(sys.exc_info(),
```

Comment:
I'm curious if the name `BadThing` is still accurate after it's been overloaded at least for mode-switching.